### PR TITLE
feat: quiz question and quiz type guards with quiz result added

### PR DESCRIPTION
**Pull request by mehulsatardekar**
files to check 👍 
src/components/quiz-question/QuizMain.tsx
 src/components/quiz-question/QuizQuestions.tsx
 src/components/quiz-question/QuizScreen.tsx
 src/components/quiz-question/quizQuestionTypes.ts
 src/pages/quiz-result/QuizResultPage.tsx
src/pages/quiz-result/quiz-result-page.css
 src/types/quiztypes.ts

user credentials: 
email : test@gmail.com
password: mehulSATARDEKAR1!

live url 👍 
https://storied-croissant-872b39.netlify.app/dashboard

### CODE REVIEW

File: src/components/quiz-question/QuizQuestions.tsx [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+         <div className="progress-bar" ref={progressBar}></div>
+
+        <div className="ui-card-text card-mb-1 ">
+          <section className="flex flex-justify-center width-full ">
+              <div className="flex flex-column gap-2 pt-1 ">
+                  <span className="font-bold font-weight-800 font-sm line-space-2">{Question.question}</span>
+                   {
+                     Question.option.map((option:string, index:number)=>{
+                       return(
+                    <label htmlFor={`ids_ ${index}`} className="btn-outline-primary btn btn-py-1 flex flex-justify-center gap">{option}
```
Comment (rahulyadav139) [inline src/components/quiz-question/QuizQuestions.tsx:72]: Suggestion: I think you forgot to write "key" attribute here.

File: src/components/quiz-question/QuizScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+     
+      getQuizQuestions()
+    },[])
+
+    useEffect(()=>{
+      if(quizQuestions.length > 0){
+
+        if(isQuenstionsEnd){
+        
+         quizScoreResult = quizQuestions.reduce((acc, curr,index) => {
```
Comment (rahulyadav139) [inline src/components/quiz-question/QuizScreen.tsx:61]: Suggestion: you should write a specific name to the accumulator and current element.

File: src/components/quiz-question/QuizScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        if(curr.answerIndex === markedAnswers[index]){
+          correctAnswers ++;
+            acc = acc+ curr.points;
+        }
+        return acc;
+       }, 0)
+
+       
+        if(quizScoreResult * (100/ quizDetails?.quiz_points!) > 50){
```
Comment (rahulyadav139) [inline src/components/quiz-question/QuizScreen.tsx:70]: Appreciation: you have used "bang operator" for some code reduction. Great work.

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/quiz-question/QuizScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+    let percentage =0;
+    let isQuizPassed : boolean;
+
+   const isQuenstionsEnd = (questionCurrentIndex === quizQuestions.length);
+  
+    const {playquizid} = useParams();
+    const {notifyError} = useToast();
+    const navigate =  useNavigate();
+    const { userDetails, getUserDetails, setUserDetails} = useAchievementDetail();
+    //const {isQuizStarted, setIsQuizStarted} = useQuizPlay();
```
Comment (gauravsinhaweb) [inline src/components/quiz-question/QuizScreen.tsx:28]: .

File: src/components/quiz-question/QuizQuestions.tsx [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    (progressBar.current!).classList.remove("active");
+
+    setTimeout(()=>{
+      (progressBar.current!).classList.add("active")
+     },0)
+  }
+   
+
+
+   //clearInterval(timer);
```
Comment (gauravsinhaweb) [inline src/components/quiz-question/QuizQuestions.tsx:37]: clear commented code

**Review by gauravsinhaweb — APPROVED** [review]

**Pull request merged**